libpod/container.go

```diff
@@ -1173,6 +1173,58 @@ func (c *Container) Networks() ([]string, bool, error) {
 	return c.networks()
 }
 
+// NetworkMode gets the configured network mode for the container.
+// Get actual value from the database
+func (c *Container) NetworkMode() (string, error) {
+	var err error
+	networkMode := ""
+	ctrSpec := c.config.Spec
+
+	// fetch spec from existing state if nil provided.
+	if ctrSpec == nil {
```

Comment:
This will never be true, you can drop the check - we will never create a container with a nil state.

Reply:
Further - we create the spec that `specFromState()` would retrieve *from* `c.config.Spec` so that will also fail if `c.config.Spec` does not exist

Reply:
@mheon thanks, I have removed the check in latest commit.